CMakeLists.txt

```diff
@@ -171,6 +176,11 @@ else (USE_EXT_GD)
 		SET(HAVE_LIBWEBP 1)
 	ENDIF(WEBP_FOUND)
 
+	IF(AVIF_FOUND)
+		INCLUDE_DIRECTORIES(${AVIF_INCLUDE_DIR})
```

Comment:
ok! let me try this and make sure I can still build.

Reply:
@cryptomilk, I've had trouble getting this to work. I think we need your superior experience with CMake here!

Looking into this deeply - `libavif-config.cmake` contains the line `add_library(avif SHARED IMPORTED)`. That seems promising. But I don't know the mechanism by which that library gets added. 

In my current build process, `find_library()` locates `libavif` and `aom` and adds the relevant info to `AVIF_LIBRARIES`. These then propagate to `LIBGD_DEP_LIBS`, which get added to the list of things to link by `target_link_libraries(${GD_LIB} ${LIBGD_DEP_LIBS})`.

I'm not sure how to replicate that process with `libavif-config.cmake`.

Would you consider fixing this yourself in a followup PR?

Reply:
Did you use `find_package(libavif 0.8.2 CONFIG)` and remove the `cmake/Modules/FindAVIF.cmake` file?

All you have to do is:

```
if (TARGET avif)
  append(LIBGD_DEP_LIBS avif)
endif()
```

Reply:
I did indeed use `find_package(libavif 0.8.2 CONFIG REQUIRED)` and remove `cmake/Modules/FindAVIF.cmake`!

I can try this - but what about `aom`? The build needs to find and link that library too.

Reply:
Why would it need to link aom?

Reply:
I'd included that - but maybe it's unnecessary.

I just tried your suggestion - thanks for that! I had to amend it, though:

```
if (TARGET avif)
	LIST(APPEND LIBGD_DEP_LIBS avif)
endif()
```

Unfortunately this just appended the word `avif` to the list of libraries:

`.../usr/local/lib/libwebp.a;avif`

I think it would save us both several more hours if you were willing to make this change via a followup PR :)

Reply:
This looks correct, `avif` is a cmake target!

Reply:
Sadly, it just appended the word `avif` to the list of libraries. It didn't work for me 😭

Reply:
Thanks for submitting that patch. That... does work for me!

😅